[RFC] SOTA research appendix

FlorentRevest opened the pull request:
This is my first attempt at detailing my researches on SOTA solutions for PELUX. This might need a couple of additional iterations depending on the comments you have.

FlorentRevest committed:
SOTA Client: Detail SOTA case studies in an Appendix

Signed-off-by: Florent Revest <revestflo@gmail.com>

FlorentRevest committed:
Dictionnary: Adds some keywords to the aspell.en vocabulary so that the SOTA paragraphs can pass test

Signed-off-by: Florent Revest <revestflo@gmail.com>

tobsan commented on docs/chapters/architecture/SOTA-client.rst:
```diff
@@ -64,3 +64,115 @@ Glossary
+following paragraphs analyzes various update solutions in the specific context
+of PELUX.
+
+**Sysup**: This is a very simple tool to achieve A/B partition switching. It is
```
I would prefer a sub-heading here for each alternative here.

tobsan commented on docs/chapters/architecture/SOTA-client.rst:
```diff
@@ -64,3 +64,115 @@ Glossary
+**OSTree**: OSTree is an elegant file-based update mechanism that uses hard links
+to achieve in-place(no A/B partitioning) atomic updates. It is often described
+as a "git for operating systems". It currently benefits from a very large and
+active community. It has support for rollback  It integrates with Yocto easily.
```
Missing .

tobsan commented on docs/chapters/architecture/SOTA-client.rst:
```diff
@@ -64,3 +64,115 @@ Glossary
+implementation is Aktualizr, can download any kind of data from the server and
+relay that to an installer. It is also worth noting that this implementation
+supports complex security mechanisms using Uptane and RVI. This architecture
+isn't enough on its own but can be integrated with something else to provide a
```
Why isn't it enough on its own? It doesn't follow naturally from the rest of the text.

tobsan commented on docs/chapters/architecture/SOTA-client.rst:
```diff
@@ -64,3 +64,115 @@ Glossary
+
+**QtOTA**: This solution contains a set of scripts and QML APIs to easily
+integrate OSTree in a Yocto and Qt/QML system. QtOTA seems preferable over
+OSTree alone if the final system is tightly linked to a Qt architecture.
```
Given, I assume, that we don't need to update bootloaders or MCUs?

tobsan commented on docs/chapters/architecture/SOTA-client.rst:
```diff
@@ -64,3 +64,115 @@ Glossary
+with complex fleet management systems such as Hawkbit, it guarantees integrity
+and authentication, offers APIs for GUI integration, is easily integrated to
+Yocto and can be extended with handlers to upgrade FPGAs, MCUs or other
+components and is well documented... SWUpdate meets the requirements of PELUX.
```
I think a single period suffices here :)

tobsan commented on docs/chapters/architecture/SOTA-client.rst:
```diff
@@ -64,3 +64,115 @@ Glossary
+components and is well documented... SWUpdate meets the requirements of PELUX.
+
+**RAUC**: This solution is failsafe, atomic, can revert to a previous state, is
+flexible enough when it comes to partition layout, uses bundle of images that
```
"uses a bundle"

tobsan commented on docs/chapters/architecture/SOTA-client.rst:
```diff
@@ -64,3 +64,115 @@ Glossary
+
+**RAUC**: This solution is failsafe, atomic, can revert to a previous state, is
+flexible enough when it comes to partition layout, uses bundle of images that
+can be downloaded from the network or from local media, interfaces with Hawkbit,
```
Perhaps a link to Hawkbit, or a description of what it is, would be good

jeremiah replied:
Might be worth looking at https://advancedtelematic.github.io/rvi_sota_server/ since it is purpose built for automotive and modular.

FlorentRevest replied:
rvi_sota_server/client haven't been updated for a long while now https://github.com/genivi/rvi_sota_client and it looks like RVI is abandoned.
You can check the "GENIVI SOTA (Aktualizr)" paragraph for some more details.
Also, we finished the Architecture Meeting with the conclusion that we wouldn't rely on GENIVI SOTA.

GordanM replied:
Hey @jeremiah , thanks for brining this up. I believe that the github link to that work would be this one: https://advancedtelematic.github.io/rvi_sota_server/. Like @FlorentRevest mentioned, the efforts were abandoned. We can always bring back the topic and integrate our solution against the RVI if we find a future scenario for it.

jeremiah replied:
The RVI efforts were abandoned by ATS, but RVI is being used in production by JLR. ATS is moving to use TLS directly, which I think is smart. In any case, I just wanted to put GENIVI's SOTA on the radar but since you've already investigated, I think we can skip it to keep the list shorter.

tobsan reviewed (changes requested):
Links to each of these alternatives would be nice, otherwise I had mostly cosmetic comments.

GordanM commented:
@FlorentRevest great work! Just a small request, what @tobsan mentioned already: It would be great to have references and links to the update solutions and other technologies like Hawkbit, etc. You can take a look at: http://www.sphinx-doc.org/en/stable/rest.html#hyperlinks.

FlorentRevest commented:
I updated my PR according to your comments.

tobsan commented:
@FlorentRevest check the Jenkins output, it complains about a "misspelled" word.

FlorentRevest commented:
Could I have more info on the version of aspell-en that is used on the CI server? It's the seventh times I meet a word that is correctly recognized on my devel platform (latest stable fedora) but not on Jenkins.

Otherwise, I fixed this word and also added a couple of references, as discussed in https://github.com/Pelagicore/software-factory-blueprint/pull/73#issuecomment-356538818
